jboss/container/jolokia/7/module.yaml

```diff
@@ -0,0 +1,66 @@
+schema_version: 1
+
+name: jboss.container.jolokia
+version: '7'
+description: ^
+  Provides support for configuring Jolokia.  Basic usage is
+  opts="$JBOSS_CONTAINER_JOLOKIA_MODULE/jolokia-opts"
+
+labels:
+- name: io.fabric8.s2i.version.jolokia
+  value: "1.6.2-redhat-00002"
+
+envs:
+- name: JOLOKIA_VERSION
+  description: Version of Jolokia being used.
+  value: "1.6.2"
+- name: AB_JOLOKIA_PASSWORD_RANDOM
+  description: Determines if a random AB_JOLOKIA_PASSWORD be generated. Set to **true** to generate random password. Generated value will be written to `/opt/jolokia/etc/jolokia.pw`.
+  value: "true"
+- name: AB_JOLOKIA_AUTH_OPENSHIFT
+  description: Switch on client authentication for OpenShift TLS communication. The value of this parameter can be a relative distinguished name which must be contained in a presented client's certificate. Enabling this parameter will automatically switch Jolokia into https communication mode. The default CA cert is set to `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt`.
+  value: "true"
+- name: AB_JOLOKIA_HTTPS
+  description: Switch on secure communication with https. By default self signed server certificates are generated if no `serverCert` configuration is given in **AB_JOLOKIA_OPTS**.
+  value: "true"
+- name: AB_JOLOKIA_OFF
+  description: If set disables activation of Joloka (i.e. echos an empty value). By default, Jolokia is enabled.
+  example: "true"
+- name: AB_JOLOKIA_CONFIG
+  description: If set uses this file (including path) as Jolokia JVM agent properties (as described in Jolokia's link:https://www.jolokia.org/reference/html/agents.html#agents-jvm[reference manual]). If not set, the `/opt/jolokia/etc/jolokia.properties` will be created using the settings as defined in the manual. Otherwise the rest of the settings in this document are ignored.
+  example: "/opt/jolokia/custom.properties"
+- name: AB_JOLOKIA_HOST
+  description: Host address to bind to. Defaults to **0.0.0.0**.
+  example: "127.0.0.1"
+- name: AB_JOLOKIA_PORT
+  description: Port to listen to. Defaults to **8778**.
+  example: "5432"
+- name: AB_JOLOKIA_USER
+  description: User for basic authentication. Defaults to **jolokia**.
+  example: "myusername"
+- name: AB_JOLOKIA_PASSWORD
+  description: Password for basic authentication. By default authentication is switched off.
+  example: "mypassword"
+- name: AB_JOLOKIA_ID
+  description: Agent ID to use (`$HOSTNAME` by default, which is the container id).
+  example: "openjdk-app-1-xqlsj"
+- name: AB_JOLOKIA_DISCOVERY_ENABLED
+  description: Enable Jolokia discovery. Defaults to **false**.
+  example: "true"
+- name: AB_JOLOKIA_OPTS
+  description: Additional options to be appended to the agent configuration. They should be given in the format `key=value,key=value,...`.
+  example: "backlog=20"
+- name: JBOSS_CONTAINER_JOLOKIA_MODULE
```

Comment:
Not to be annoying, but this looks like the same API question as prometheus. We've documented it, so I think we should be careful in before breaking it. Should we not leave the variable in above and use /usr/share/java/jolokia-jvm-agent/ here, allowing anyone who did change this for whatever reason to not be broken by this change?

Reply:
In both cases, would you be satisfied by a compatibility symlink from the old location (under e.g. `JBOSS_CONTAINER_JOLOKIA_MODULE`) to the new one?